functests/dpdk/dpdk.go

```diff
@@ -299,10 +319,54 @@ func tryToFindDPDKPod() (*corev1.Pod, bool) {
 	return nil, false
 }
 
-func findOrOverridePerformanceProfile() {
-	performanceProfile, valid, err := ValidatePerformanceProfile()
-	Expect(err).ToNot(HaveOccurred())
+func discoverPerformanceProfile() (bool, string) {
+	reason := "Can not run tests in discovery mode. Failed to find a valid perfomance profile. %s"
+
+	if enforcedPerformanceProfileName != "" {
+		performanceProfile, err := findDefaultPerformanceProfile()
+		if err != nil {
+			return false, fmt.Sprintf(reason, err)
+		}
+		valid, err := ValidatePerformanceProfile(performanceProfile)
+		if !valid || err != nil {
+			return false, fmt.Sprintf(reason, err)
+		}
+		return true, ""
+	}
 
+	performanceProfileList := &perfv1alpha1.PerformanceProfileList{}
+	err := client.Client.List(context.TODO(), performanceProfileList)
+	if err != nil {
+		return false, fmt.Sprintf(reason, err)
+	}
+	for _, performanceProfile := range performanceProfileList.Items {
```

Comment:
You may need to return this. The ultimate goal is making sure that we are running a pod where it can run.
To drive it, we have three things to influence it:
- the resources we ask (that may fit the perf profile)
- the network resource
- the node selector
I would mix all of them while doing the discovery, to make sure the dpdk pod is scheduled on the right node.

If we handle them separately, there can be a valid performance profile impacting a set of nodes which is different from the set of nodes containing a valid sriov device.